packages/licenses/src/getPkgInfo.ts

```diff
@@ -0,0 +1,356 @@
+ */
+const LICENSE_FILES = [
+  'LICENSE',
+  'LICENCE',
+  'LICENSE.md',
+  'LICENCE.md',
+  'LICENSE.txt',
+  'LICENCE.txt',
+  'MIT-LICENSE.txt',
+  'MIT-LICENSE.md',
+  'MIT-LICENSE',
+]
+
+export interface LicenseInfo {
+  name: string
+  licenseFile?: string
+}
+
+/**
+ * Coerce the given value to a string or a null value
+ * @param field the string to be converted
+ * @returns string | null
+ */
+function coerceToString (field: unknown): string | null {
+  const string = String(field)
+  return typeof field === 'string' || field === string ? string : null
+}
+
+/**
+ * Parses the value of the license-property of a
+ * package manifest and return it as a string
+ * @param field the value to parse
+ * @returns string
+ */
+function parseLicenseManifestField (field: unknown) {
+  if (Array.isArray(field)) {
+    const licenses = field
+    const licenseTypes = licenses.reduce((listOfLicenseTypes, license) => {
+      const type = coerceToString(license.type) ?? coerceToString(license.name)
+      if (type) {
+        listOfLicenseTypes.push(type)
+      }
+      return listOfLicenseTypes
+    }, [])
+
+    if (licenseTypes.length > 1) {
+      const combinedLicenseTypes = licenseTypes.join(' OR ') as string
+      return `(${combinedLicenseTypes})`
+    }
+
+    return licenseTypes[0] ?? null
+  } else {
+    return (field as { type: string })?.type ?? coerceToString(field)
+  }
+}
+
+/**
+ * Reads the license field or LICENSE file from
+ * the directory of the given package manifest
+ *
+ * If the package.json file is missing the `license`-property
+ * the root of the manifest directory will be scanned for
+ * files named listed in the array LICENSE_FILES and the
+ * contents will be returned.
+ *
+ * @param {*} pkg the package to check
+ * @returns Promise<LicenseInfo>
+ */
+async function parseLicense (
+  pkg: {
+    manifest: PackageManifest
+    files:
+    | { local: true, files: Record<string, string> }
+    | { local: false, files: Record<string, PackageFileInfo> }
+  },
+  opts: { cafsDir: string }
+): Promise<LicenseInfo> {
+  let licenseField: unknown = pkg.manifest.license
+  if ('licenses' in pkg.manifest) {
+    licenseField = (
+      pkg.manifest as PackageManifest & {
+        licenses: unknown
+      }
+    ).licenses
+  }
+  const license = parseLicenseManifestField(licenseField)
+
+  // check if we discovered a license, if not attempt to parse the LICENSE file
+  if (!license || /see license/i.test(license)) {
+    for (const filename of LICENSE_FILES) {
+      try {
+        let licenseContents: Buffer | undefined
+        if (pkg.files.local) {
+          const licensePackageFileInfo = pkg.files.files[filename]
+          licenseContents = await readFile(licensePackageFileInfo)
+        } else {
+          const licensePackageFileInfo = pkg.files.files[filename]
+          licenseContents = await readLicenseFileFromCafs(
+            opts.cafsDir,
+            licensePackageFileInfo.integrity
+          )
```

Comment:
Why are you not checking whether the filename is in the files? Now this will of course fail because the file info will be undefined. If you will check it, you can remove the try/catch

```javascript
        if (pkg.files.local) {
          const licensePackageFileInfo = pkg.files.files[filename]
          if (!licensePackageFileInfo) continue
          licenseContents = await readFile(licensePackageFileInfo)
        } else {
          const licensePackageFileInfo = pkg.files.files[filename]
          if (!licensePackageFileInfo) continue
          licenseContents = await readLicenseFileFromCafs(
            opts.cafsDir,
            licensePackageFileInfo.integrity
          )
```

Reply:
Moved  the file existence check outside of this if-statement